src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs

```diff
@@ -230,21 +230,31 @@ public void InitializeService(IProtocolEndpoint serviceHost)
                     if ((sqlConn != null) && !connInfo.IsSqlDW && !connInfo.IsAzure)
                     {
                         BackupOperation backupOperation = DisasterRecoveryService.Instance.SetBackupInput(helper.DataContainer, sqlConn, backupParams.BackupInfo);
+                        SqlTask sqlTask = null;
 
                         // create task metadata
                         TaskMetadata metadata = new TaskMetadata();
                         metadata.ServerName = connInfo.ConnectionDetails.ServerName;
                         metadata.DatabaseName = connInfo.ConnectionDetails.DatabaseName;
-                        metadata.Name = SR.Backup_TaskName;
-                        metadata.IsCancelable = true;
                         metadata.Data = backupOperation;
-                        
-                        // create backup task and perform
-                        SqlTask sqlTask = SqlTaskManager.Instance.CreateTask(metadata, Instance.BackupTaskAsync);
-                        sqlTask.Run();
+                        metadata.IsCancelable = true;
+
+                        if (backupParams.IsScripting)
+                        {
+                            metadata.Name = string.Format("{0} {1}", SR.BackupTaskName, SR.ScriptTaskName);
+                            metadata.TaskExecutionMode = TaskExecutionMode.Script;
+                            sqlTask = SqlTaskManager.Instance.CreateTask(metadata, Instance.BackupScriptTaskAsync);
+                            sqlTask.Run();
+                        }
+                        else
+                        {
+                            metadata.Name = SR.BackupTaskName;
```

Comment:
Didn't we discuss defaulting to "Execute and generate script" in this case? E.g. if you're executing, why not still script first then execute so users can review what happened?